src/sinks/aws_s3/sink.rs

```diff
@@ -71,19 +75,26 @@ impl RequestBuilder<(String, Vec<Event>)> for S3RequestOptions {
                 .unwrap_or_else(|| formatted_ts.to_string())
         };
 
+        let ssekms_key_id = metadata.partition_key.ssekms_key_id.clone();
+        let mut s3_options = self.api_options.clone();
+        s3_options.ssekms_key_id = ssekms_key_id;
+
         let extension = self
             .filename_extension
             .as_ref()
             .cloned()
             .unwrap_or_else(|| self.compression.extension().into());
-        metadata.partition_key = format!("{}{}.{}", metadata.partition_key, filename, extension);
+
+        metadata.s3_key = format!("{}{}.{}", metadata.s3_key, filename, extension);
+
+        debug!("S3Request s3_key: {:?}", metadata.s3_key);
```

Comment:
How serious would it be for this to leak into the public eye? I guess it's just the ID rather than the actual key? I'm just thinking it might still be best to not risk it leaking into the logs..

Reply:
Yeah, it's an ID for the key...but that's a fair point.  It was really only helping me during development/testing.  I'll remove it